test/controllers/actions/MessageTransformerSpec.scala

```diff
@@ -33,63 +30,84 @@ import models.UnloadingRemarksRejectedResponse
 import models.XMLSubmissionNegativeAcknowledgementResponse
 import models.request.ArrivalRequest
 import org.scalacheck.Arbitrary.arbitrary
-import org.scalacheck.Gen
+import org.scalatest.EitherValues
 import org.scalatest.OptionValues
 import org.scalatest.concurrent.ScalaFutures
 import org.scalatest.freespec.AnyFreeSpec
 import org.scalatest.matchers.must.Matchers
 import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
 import play.api.mvc.Results.Ok
 import play.api.mvc._
+import play.api.test.DefaultAwaitTimeout
 import play.api.test.FakeRequest
+import play.api.test.FutureAwaits
 import play.api.test.Helpers._
 import play.twirl.api.HtmlFormat
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
-
-class InboundMessageTransformerSpec extends AnyFreeSpec with Matchers with ScalaCheckPropertyChecks with ModelGenerators with OptionValues with ScalaFutures {
-
-  private val arrival: Arrival     = arbitrary[Arrival].sample.value
-  private val arrivalId: ArrivalId = arrival.arrivalId
-
-  private val successfulResponse: Request[AnyContent] => Future[Result] = {
+import scala.xml.NodeSeq
+
+class MessageTransformerSpec
+    extends AnyFreeSpec
+    with Matchers
+    with ScalaCheckPropertyChecks
+    with ModelGenerators
+    with OptionValues
+    with EitherValues
+    with FutureAwaits
+    with DefaultAwaitTimeout
+    with ScalaFutures {
+
+  private val arrival: Arrival = arbitrary[Arrival].sample.value
+
+  private val successfulResponse: Request[NodeSeq] => Future[Result] = {
     _ =>
       Future.successful(Ok(HtmlFormat.empty))
   }
 
-  private def action = new InboundMessageTransformer()
-
-  private val xmlNegativeAcknowledgement: ArrivalStatus =
-    Gen.oneOf(Seq(ArrivalXMLSubmissionNegativeAcknowledgement, UnloadingRemarksXMLSubmissionNegativeAcknowledgement)).sample.value
+  private lazy val action = new MessageTransformer()
 
   "InboundMessageTransformer" - {
 
-    "returns 200 for supported `X-Message-Type`" in {
+    "returns 200 for supported root node" in {
       forAll(arbitrary[ChannelType]) {
         channel =>
           val arrivalMovement = arrival.copy(status = ArrivalStatus.ArrivalSubmitted)
 
           val request =
-            ArrivalRequest(FakeRequest("", "").withHeaders("X-Message-Type" -> MessageType.ArrivalRejection.code), arrivalMovement, channel)
+            ArrivalRequest(
+              FakeRequest("", "")
+                .withHeaders(
+                  "X-Message-Type" -> MessageType.ArrivalRejection.code,
+                  "Content-Type"   -> "application/xml"
+                )
+                .withBody[NodeSeq](<CC008A> </CC008A>),
```

Comment:
Is it worth adding a Gen here for the rootnodes to test them all (we have them available in the MessageType model)

Reply:
This also depends on the status of the arrival so might be a complex generator. will test all of them manually